inference-engine/src/transformations/src/ngraph_ops/eltwise.cpp

```diff
@@ -37,16 +37,23 @@ void op::Eltwise::validate_and_infer_types() {
     NODE_VALIDATION_CHECK(this, element::Type::merge(et_result, data1_et, data2_et),
                           "Element types for first and second do not match :", data1_et, " and ", data2_et);
 
-    auto shape1 = get_input_partial_shape(0).to_shape();
-    auto shape2 = get_input_partial_shape(1).to_shape();
+    if (get_input_partial_shape(0).rank().is_dynamic() ||
+        get_input_partial_shape(1).rank().is_dynamic()) {
+        set_output_type(0, et_result, PartialShape::dynamic());
+        return;
+    }
+
+    std::vector<ngraph::Dimension> shape1(get_input_partial_shape(0));
+    std::vector<ngraph::Dimension> shape2(get_input_partial_shape(1));
 
-    ngraph::Shape output_shape(std::max(shape1.size(), shape2.size()));
+    std::vector<ngraph::Dimension> output_shape(ngraph::PartialShape::dynamic(std::max(shape1.size(), shape2.size())));
     auto output_shape_it = output_shape.rbegin();
 
     auto shape1_it = shape1.rbegin(), shape2_it = shape2.rbegin();
     while (shape1_it != shape1.rend() || shape2_it != shape2.rend()) {
-        if (shape1_it != shape1.rend() && shape2_it != shape2.rend()) {
-            *output_shape_it = std::max(*shape1_it, *shape2_it);
+        if (shape1_it != shape1.rend() && shape2_it != shape2.rend() &&
```

Comment:
I think that the behavior here will be wrong if `shape1_it != shape1.rend() && shape2_it != shape2.rend()` but only one of the elements is static.

Reply:
Yes, thanks! I'll fix that.

Reply:
Done.